src/module.cpp

```diff
@@ -901,7 +901,10 @@ generic_split(const module& m,
                    splits.end(),
                    std::back_inserter(outputs),
                    [&](instruction_ref ins) { return map_ins1.at(ins); });
-    m1.add_return(outputs);
+    if(not outputs.empty())
+    {
+        m1.add_return(outputs);
+    }
```

Comment:
It could also occur when `split` instruction is not found in the module that is being split. 
For example : 
case : {gemm + reshape + pointwise} --> in this case original module is first split with gemm instruction as the split.
so it becomes {gemm} + {reshape + pointwise} and then 
{reshape + pointwises} is split with reshape_ins as the split instruction. 

But for the case where we only have `{gemm + pointwise}` in this case 

First split would do `{gemm} + {Pointwise}` later second split would try to split `{pointwise}` with `std::prev{pointwise}` as the split instruction that would be `gemm` and it won't be found in `{pointwise}`.

Reply:
Reverted these changes. Marking this comment as resolved.